### PR TITLE
probably done with ncit mini syllabus

### PR DESCRIPTION
**Pull request by sheenarbw**
Related issues: [please specify]

## Description:

content/syllabuses/ncit-javascript.md:
These are the cards that get generated in order. If this is fine then making the syllabuses for the others (python,  java) would be easy peasy
```
T: Big Picture of Web Dev - flavours [] - tags ['todo', 'internet'] http://syllabus.africacode.net/workshops/big-picture-web-dev/
T: How the internet works - flavours [] - tags ['ncit', 'internet'] http://syllabus.africacode.net/national-qualifications-framework/ncit/how-the-internet-works/
T: Beginning Bash - flavours [] - tags ['bash'] http://syllabus.africacode.net/topics/bootcamp/bash-for-bootcamp/
T: Git Basics - flavours [] - tags ['git'] http://syllabus.africacode.net/topics/git/
T: Basic introduction to Github - flavours [] - tags ['git', 'github'] http://syllabus.africacode.net/topics/github/intro/
P: Git Basic Exercises - flavours [] - tags ['git'] http://syllabus.africacode.net/projects/git-exercises/
T: Introduction to assertive programming - flavours [] - tags ['defensive-programming'] http://syllabus.africacode.net/topics/intro-to-asserive-programming/
P: Morse code - flavours [<Tag: javascript>] - tags ['data structures', 'problem solving'] http://syllabus.africacode.net/projects/morse-code/
T: Unit testing (language agnostic concepts) - flavours [] - tags ['unit testing'] http://syllabus.africacode.net/topics/unit-testing/
T: Test Driven Development - flavours [] - tags ['tdd'] http://syllabus.africacode.net/topics/test-driven-development/
P: simple-calculator part 1 - flavours [<Tag: javascript>] - tags ['tdd'] http://syllabus.africacode.net/projects/tdd/simple-calculator-part1/
T: Version Control - flavours [<Tag: javascript>] - tags ['ncit', 'git', 'clean code'] http://syllabus.africacode.net/national-qualifications-framework/ncit/version-control/
T: Clean Code - flavours [] - tags ['clean code'] http://syllabus.africacode.net/topics/clean-code/
T: Pseudocode and documentation - flavours [] - tags ['todo', 'ncit', 'documentation'] http://syllabus.africacode.net/national-qualifications-framework/ncit/pseudocode-and-documentation/
T: Problem decomposition - flavours [] - tags ['problem solving'] http://syllabus.africacode.net/language-agnostic/problem-solving-technique/
P: Get started with Hackerrank - flavours [<Tag: javascript>] - tags ['data structures', 'algorithms', 'problem solving'] http://syllabus.africacode.net/language-agnostic/intro-to-hackerrank/
T: Thinking like a machine - flavours [<Tag: javascript>] - tags ['ncit', 'problem-solving', 'binary-arithmetic'] http://syllabus.africacode.net/national-qualifications-framework/ncit/thinking-like-a-machine/
P: Validate a South African ID number - flavours [<Tag: javascript>] - tags ['problem-solving', 'tdd'] http://syllabus.africacode.net/language-agnostic/validate-id-number/
T: Test-driven development - flavours [<Tag: javascript>] - tags ['ncit', 'tdd'] http://syllabus.africacode.net/national-qualifications-framework/ncit/test-driven-development/
T: Constructive Feedback - flavours [] - tags ['ncit', 'communication'] http://syllabus.africacode.net/national-qualifications-framework/ncit/constructive-feedback/
P: Person - flavours [<Tag: javascript>] - tags ['oop'] http://syllabus.africacode.net/projects/oop/person/
P: Animals Part 1. OOP basics - flavours [<Tag: javascript>] - tags ['oop'] http://syllabus.africacode.net/projects/oop/animals/part1/
T: Object-Oriented Programming - flavours [<Tag: javascript>] - tags ['ncit', 'oop'] http://syllabus.africacode.net/national-qualifications-framework/ncit/object-oriented-programming/
T: Conducting Research and User Interviews - flavours [] - tags ['todo', 'ncit', 'research'] http://syllabus.africacode.net/national-qualifications-framework/ncit/conducting-research-and-user-interviews/
T: SQL - flavours [] - tags ['sql'] http://syllabus.africacode.net/sql/topics/intro-to-relational-databases/
T: Introduction to Bash and the terminal - flavours [] - tags ['bash', 'linux'] http://syllabus.africacode.net/topics/linux/intro-to-bash/
T: Environmental Variables - flavours [] - tags ['bash', 'environmental variables'] http://syllabus.africacode.net/topics/linux/os-environmental-variables/
T: Intro to Docker and Docker-compose - flavours [] - tags ['postgres', 'docker-compose'] http://syllabus.africacode.net/topics/intro-to-docker/
P: Shop Database using sql - flavours [] - tags ['sql'] http://syllabus.africacode.net/sql/shop-project/
T: Database Development - flavours [] - tags ['ncit', 'sql'] http://syllabus.africacode.net/national-qualifications-framework/ncit/database-development/
T: optimizations and big-O - flavours [] - tags ['problem solving', 'big-o'] http://syllabus.africacode.net/topics/optimizations-and-big-o/
T: Search and Sort Techniques - flavours [] - tags ['ncit', 'algorithms', 'big-o'] http://syllabus.africacode.net/national-qualifications-framework/ncit/search-and-sort-techniques/
T: Basic HTML and HTML5 - flavours [] - tags ['html', 'responsive web design', 'html5'] http://syllabus.africacode.net/topics/free-code-camp/responsive-web-design/1-basic-html-and-html-5/
T: Basic CSS - flavours [] - tags ['css', 'responsive web design'] http://syllabus.africacode.net/topics/free-code-camp/responsive-web-design/2-basic-css/
T: Introduction to web design - flavours [] - tags ['web design'] http://syllabus.africacode.net/topics/web-frontend/intro-to-web-design/
T: What to put in your CV - flavours [] - tags ['getting a job'] http://syllabus.africacode.net/topics/what-to-put-in-your-cv/
P: Build your first personal website - flavours [] - tags ['html', 'css', 'responsive web design'] http://syllabus.africacode.net/projects/first-personal-website/
T: Building an online business - flavours [] - tags ['ncit', 'business'] http://syllabus.africacode.net/national-qualifications-framework/ncit/building-an-online-business/
T: Web Design for Business - flavours [] - tags ['ncit', 'business', 'web design'] http://syllabus.africacode.net/national-qualifications-framework/ncit/web-design-for-business/
T: Business and Technology - flavours [] - tags ['todo', 'ncit', 'business'] http://syllabus.africacode.net/national-qualifications-framework/ncit/business-and-technology/
W: Survey design - flavours [] - tags ['surveys'] http://syllabus.africacode.net/workshops/survey-design/
T: Analytics, Surveys, and Reports - flavours [] - tags ['ncit', 'analytics', 'surveys', 'reports'] http://syllabus.africacode.net/national-qualifications-framework/ncit/analytics-surveys-and-reports/
T: Agile Project Management - flavours [] - tags ['ncit', 'agile', 'scrum'] http://syllabus.africacode.net/national-qualifications-framework/ncit/agile-project-management/
T: Agile Meetings - flavours [] - tags ['ncit', 'agile', 'scrum'] http://syllabus.africacode.net/national-qualifications-framework/ncit/agile-meetings/
T: Managing the Software Development Process - flavours [] - tags ['ncit', 'agile'] http://syllabus.africacode.net/national-qualifications-framework/ncit/managing-the-software-development-process/
T: Ethics and Professionalism - flavours [] - tags ['ncit', 'ethics'] http://syllabus.africacode.net/national-qualifications-framework/ncit/ethics-and-professionalism/
T: Presenting your findings - flavours [] - tags ['todo', 'ncit', 'communication'] http://syllabus.africacode.net/national-qualifications-framework/ncit/presenting-your-findings/
T: Write a report - flavours [] - tags ['todo', 'ncit', 'communication'] http://syllabus.africacode.net/national-qualifications-framework/ncit/write-a-report/

```
## I solemnly swear that:

- [x] I ran the hugo server and looked at my changed in the browser with my own eyes
- [x] I ran the linter and there were no errors
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: content/national-qualifications-framework/ncit/analytics-surveys-and-reports/_index.md [hunk 1]
```diff
@@ -3,27 +3,35 @@ _db_id: 372
...
 ---
 
+Collecting and understanding data is a critical skill for any kind of coding professional. 
+
 ## 1. Google Analytics
 
 Google Analytics is a vital tool for web developers and business analysts. 
 
-Register for the Analytics Academy and complete all units in the first three modules:
+If you are a web developer it would be worthwhile to register for the Analytics Academy and complete all units in the first three modules:
```
Comment (ry-oc) [inline content/national-qualifications-framework/ncit/analytics-surveys-and-reports/_index.md:24]: Small thing, there should be a space between `worth` and `while`

File: content/national-qualifications-framework/ncit/building-an-online-business/_index.md [hunk 2]
```diff
@@ -133,24 +118,19 @@ Copyright gives the creator of certain types of material rights to control the u
...
 
-
 #### Data Protection
 Data Protection requires that you notify the Information Commissioner what you are going to use the personal data you collect for. Out of this requirement comes the website Privacy Policy. The main purpose of these statements is to fulfil the information disclosure requirements of the Data Protection Act and inform or reassure your website visitors that the information they provide to you is going to be used appropriately. This has been the subject of much debate in recent years. If you’re unsure do research on current laws. Have a look at this article on South African data protection from 2018: https://uk.practicallaw.thomsonreuters.com/5-503-0787.
```
Comment (ry-oc) [inline content/national-qualifications-framework/ncit/building-an-online-business/_index.md:126]: Based on previous edits you would want to add a line here

File: content/national-qualifications-framework/ncit/building-an-online-business/_index.md [hunk 3]
```diff
@@ -133,24 +118,19 @@ Copyright gives the creator of certain types of material rights to control the u
...
 
-
 #### User Agreements
```
Comment (ry-oc) [inline content/national-qualifications-framework/ncit/building-an-online-business/_index.md:128]: Based on previous edits you would want to add a line here

File: content/national-qualifications-framework/ncit/building-an-online-business/_index.md [hunk 4]
```diff
@@ -133,24 +118,19 @@ Copyright gives the creator of certain types of material rights to control the u
...
 Users must check a box stating they agree to your businesses User Agreements and your Privacy Policy. They must “opt in” to these agreements when they sign up for your site or purchase a product or service from you.
 
-
 #### Email Marketing
 It is illegal to send unsolicited email messages except in limited circumstances. If customers have consented to receiving information from you in the past, i.e. opted in, you can send them information on other things you think they might be interested in. You must give these people the option to opt out of receiving any further messages from you. Your email marketing software will usually help to ensure that you conform to legal requirements.
```
Comment (ry-oc) [inline content/national-qualifications-framework/ncit/building-an-online-business/_index.md:132]: Based on previous edits you would want to add a line here

File: content/national-qualifications-framework/ncit/building-an-online-business/_index.md [hunk 5]
```diff
@@ -133,24 +118,19 @@ Copyright gives the creator of certain types of material rights to control the u
...
 
-
 #### Returned Goods
```
Comment (ry-oc) [inline content/national-qualifications-framework/ncit/building-an-online-business/_index.md:134]: Based on previous edits you would want to add a line here

File: content/national-qualifications-framework/ncit/version-control/_index.md [hunk 6]
```diff
@@ -18,96 +22,109 @@ topic_needs_review: true
...
 
+### Organising your project files
 
-Documentation for Phaser can be found at this link:
+As your software and web development projects become more complex you will create more and more files. Even for for a basic website, you have multiple HTML files, a CSS file, images, and maybe some JavaScript files. Following common industry standards on keeping your project organised will make your life as a developer so much easier. Plus, it will enable you to share your code and work with other coders more easily.
 
-https://phaser.io/docs/2.6.2/index
+There isn't really one standard to rule all standards. Different teams and organisations organise their code in different ways. Be consistent with conventioned that are already in place. And if you are starting a new project then be diligent about keeping oyur code tidy!
 
+Please refer to this project to see a bit more about how you'll be excpected to organise your code during this course:
```
Comment (ry-oc) [inline content/national-qualifications-framework/ncit/version-control/_index.md:63]: Small spelling error `excpected`

File: content/syllabuses/ncit-javascript.md [hunk 7]
```diff
@@ -1,47 +1,25 @@
 ---
+_db_id: 37
 title: NCIT - JavaScript
 ---
 
-- {{% contentlink path="national-qualifications-framework/ncit/version-control" %}}
```
Comment (ry-oc) [inline content/syllabuses/ncit-javascript.md:5]: We usually have them do these 3 first as they have just come out of the design thinking sprint and it and their research, presentations etc are fresh:
- {{% contentlink path="national-qualifications-framework/ncit/conducting-research-and-user-interviews" %}}
- {{% contentlink path="national-qualifications-framework/ncit/presenting-your-findings" %}}
- {{% contentlink path="national-qualifications-framework/ncit/write-a-report" %}}

But if this order makes more sense then I am happy

**Review by ry-oc — COMMENTED** [review]
Left a couple of comments, but looks fine over all.

Main comment is on the syllabus

File: content/syllabuses/ncit-javascript.md (hunk 7 above)
Comment (sheenarbw) [reply content/syllabuses/ncit-javascript.md:5]: I want to push to have the design sprint happen later. If it happens immediately as people join Umuzi then Make Umuzi Better is more like Make Random Assumptions About Umuzi. MRAAU just doesn't have the same ring to it.

**Pull request merged**